xen-bugtool

```diff
@@ -765,6 +769,7 @@ exclude those logs from the archive.
     cmd_output(CAP_DISK_INFO, [SG_MAP, '-x'])
     func_output(CAP_DISK_INFO, 'scsi-hosts', dump_scsi_hosts)
     cmd_output(CAP_DISK_INFO, [LVDISPLAY, '--map'])
+    cmd_output(CAP_BLOCK_SCHEDULER, [LSBLK, '-o', 'SCHED,NAME'])
```

Comment:
If you only want the block scheduler of each device, this would be the direct way:
```suggestion
    file_output(CAP_BLOCK_SCHEDULER, ["/sys/class/block/*/queue/scheduler"])
```
While I'd agree that running `[LSBLK, '-o', 'SCHED,NAME'])` doesn't take a long time on it's own, and far from the load that `cmd_output(CAP_HDPARM_T, [HDPARM, '-tT', '/dev/%s' % d])` in Dom0, I'd just like to raise the awareness of that when we can get the same info in an easer way that is also better for unit testing, I'd prefer that.

If you want the output in one file:

If we run lsblk at all, then it should be more worth doing th call than just getting the block scheduler to not have to add a second, third, .. call to lsblk to get more information.

I'd recommend to at least get als the transport (e.g. "sata"), if it's a rotational device or not, the sector size, the request-size, and vendor,model,serial.

Example:

```bash
lsblk -io type,name,sched,tran,rota,log-sec,rq-size,vendor,model,serial
TYPE NAME                                                            SCHED TRAN   ROTA LOG-SEC RQ-SIZE VENDOR   MODEL            SERIAL
disk sda                                                             cfq   sata      1     512     128 ATA      ST1000NX0423     W470MWXE
part |-sda4                                                          cfq             1     512     128                           
part |-sda2                                                          cfq             1     512     128                           
part |-sda5                                                          cfq             1     512     128                           
part |-sda3                                                          cfq             1     512     128                           
lvm  | `-VG_XenStorage--03086bb3--efa1--a74e--fbc5--16077bf518e7-MGT                 1     512     128                           
part |-sda1                                                          cfq             1     512     128                           
part `-sda6                                                          cfq             1     512     128
```

As in this example, I'd also add `-i` to use ASCII characters for the tree formatting as otherwise `lsblk` would use Unicode characters to format the tree if it runs inside a UTF-8 locale (like when running it from the shell) but ASCII format if running it from some XAPI plugin (in which the `en_US.UTF-8` default locale of XenServer) is not set. So for a clean and more-stable ASCII output, I'd recommend adding it.

Also, because the use of `,` in the arguments would also be carried over into the generated filename by `cmd_output`, strongly recommend to set the argument `label` to e.g. `label="lsblk"`, to get a file like `bug-report-20230622161233/lsblk.out`.

Otherwise a file like `bug-report-20230622161233/lsblk-osched,name.out`, would be used, where the `,` in the name which is illegal on Windows in file names.

Reply:
I started with using `/sys/class/block/*/queue/scheduler` and this has a number of problems

1. picks up all kernel block devices, including the tdX devices for the live, active VBDs attached to VM. Where there may be many hundred of these. Filtering to just `sd*` and `dm-*` would be possible but potentially risks missing something that could be important in triaging a customer issue.
2. The output results in one file in the bugtool for each entry in `/sys/class/block/*/queue` which results in a lot of files to deal with when triaging the logs whereas `lsblk` gives the result in one single file.
3. Due to the implementation of sysfs the stat call for reading the size of the logs will always report 4KB so you very rapidly (and erroneously) exceed the size limit for the capability, or the size limit has to be set artificially high.